Add(mp): Azure DevTools App Configuration (#634)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/cloud-azure-devtools-appconfiguration.md

title: Azure App Configuration
---

## Overview

Azure App Configuration helps manage application settings and control their 
access centrally by storing their configuration in a universal hosted location. 

The Centreon Plugin-Pack *Azure App Configuration* can rely on Azure API or Azure CLI 
to collect the metrics related to the App Configuration service.

## Plugin Pack Assets

### Monitored Objects

* Azure *App Configuration* HTTP requests

### Discovery rules

The Centreon Plugin-Pack *Azure App Configuration* includes a Host Discovery *provider* to automatically discover the Azure instances of a given
subscription and add them to the Centreon configuration.
This provider is named **Microsoft Azure App Configuration**:

![image](../../../assets/integrations/plugin-packs/procedures/cloud-azure-devtools-appconfiguration-provider.png)

> This discovery feature is only compatible with the 'api' custom mode. 'azcli' is not supported yet.

More information about the Host Discovery module is available in the Centreon documentation:
[Host Discovery](../../../monitoring/discovery/hosts-discovery.html)

### Collected Metrics

<!--DOCUSAURUS_CODE_TABS-->

<!--Http-Requests-->

> | Metric Name                                             | Description                            |
> |:--------------------------------------------------------|:---------------------------------------|
> | appconfiguration.http.incoming.requests.count           | Total number of incoming http requests |
> | appconfiguration.http.incoming.requests.milliseconds    | Latency on an http request             |
> | appconfiguration.http.throttled.incoming.requests.count | Throttled http requests                |

Reviewer: ```suggestion
| Metric Name                                             | Description                            | Unit  |
|:--------------------------------------------------------|:---------------------------------------|:------|
| appconfiguration.http.incoming.requests.count           | Total number of incoming http requests | Count |
| appconfiguration.http.incoming.requests.milliseconds    | Latency on an http request             | ms    |
| appconfiguration.http.throttled.incoming.requests.count | Throttled http requests                | Count |
```

2. On the Centreon Web interface, install the *Azure App Configuration* Centreon Plugin-Pack on the "Configuration > Plugin Packs > Manager" page

<!--Offline IMP License-->

1. Install the Centreon Plugin package on every Centreon poller expected to monitor Azure App Configuration resources:

```bash
yum install centreon-plugin-Cloud-Azure-DevTools-AppConfiguration-Api
```

2. Install the Centreon Plugin-Pack RPM on the Centreon Central server:

```bash
yum install centreon-pack-cloud-azure-devtools-appconfiguration.noarch
```

3. On the Centreon Web interface, install the *Azure App Configuration* Centreon Plugin-Pack on the "Configuration > Plugin Packs > Manager" page

<!--END_DOCUSAURUS_CODE_TABS-->

## Configuration

### Host

* Log into Centreon and add a new Host through "Configuration > Hosts".
* In the *IP Address/FQDN* field, set the following IP address: '127.0.0.1'.

* Select the *Cloud-Azure-DevTools-AppConfiguration-custom* template to apply to the Host.
* Once the template applied, some Macros marked as 'Mandatory' hereafter have to be configured.
These mandatory Macros differ regarding the custom mode used:

<!--DOCUSAURUS_CODE_TABS-->

<!--Azure Monitor API-->

> | Mandatory | Nom               | Description                   |
> |:----------|:------------------|:------------------------------|
> | X         | AZURECUSTOMMODE   | Custom mode 'api'             |
> | X         | AZURESUBSCRIPTION | Subscription ID               |
> | X         | AZURETENANT       | Tenant ID                     |
> | X         | AZURECLIENTID     | Client ID                     |
> | X         | AZURECLIENTSECRET | Client secret                 |
> | X         | AZURERESOURCE     | Id of the App Configuration instance |

Reviewer: ```suggestion
| Mandatory | Nom               | Description                          |
|:----------|:------------------|:-------------------------------------|
| X         | AZURECUSTOMMODE   | Custom mode 'api'                    |
| X         | AZURESUBSCRIPTION | Subscription ID                      |
| X         | AZURETENANT       | Tenant ID                            |
| X         | AZURECLIENTID     | Client ID                            |
| X         | AZURECLIENTSECRET | Client secret                        |
| X         | AZURERESOURCE     | Id of the App Configuration instance |
```

<!--Azure AZ CLI-->

> | Mandatory | Nom               | Description                   |
> |:----------|:------------------|:------------------------------|
> | X         | AZURECUSTOMMODE   | Custom mode 'azcli'           |
> | X         | AZURESUBSCRIPTION | Subscription ID               |
> | X         | AZURERESOURCE     | Id of the App Configuration instance |

Reviewer: ```suggestion
| Mandatory | Nom               | Description                          |
|:----------|:------------------|:-------------------------------------|
| X         | AZURECUSTOMMODE   | Custom mode 'azcli'                  |
| X         | AZURESUBSCRIPTION | Subscription ID                      |
| X         | AZURERESOURCE     | Id of the App Configuration instance |
```

```bash
/usr/lib/centreon/plugins/centreon_azure_devtools_appconfiguration_api.pl \
    --plugin=cloud::azure::devtools::appconfiguration::plugin \
    --mode=http-requests \
    --custommode=api \
    --subscription='xxxxxxxxx' \
    --tenant='xxxxxxxxx' \
    --client-id='xxxxxxxxx' \
    --client-secret='xxxxxxxxx' \
    --resource='APPCONF01' \
    --resource-group='xxxxxxxxx' \
    --timeframe='900' \
    --interval='PT5M' \
    --aggregation='Count' \
    --warning-http-requests='800' \
    --critical-http-requests='900'
```

Expected command output is shown below:

```bash
OK: Instance 'APPCONF01' Statistic 'count' Metrics Incoming HTTP requests: 0.00, 
Throttled Incoming HTTP requests: 0.00, Incoming HTTP requests duration: 0.00ms | 
'APPCONF01~count#http.incoming.requests.count'=0.00;0:800;0:900;0; 
'APPCONF01~count#http.throttled.incoming.requests.count'=0.00;;;0; 
'APPCONF01~count#http.incoming.requests.milliseconds'=0.00ms;;;0;
```

The command above checks the number of *http-requests* of an Azure *App Configuration* instance using the 'api' custom-mode
(```--plugin=cloud::azure::devtools::appconfiguration::plugin --mode=http-requests --custommode=api```).

> This App Configuration instance is identified by its id (```--resource='SVC001ABCD'```) and the authentication parameters

Reviewer: ```suggestion
This App Configuration instance is identified by its id (```--resource='APPCONF01'```) and the authentication parameters
```

to be used with the custom mode are specified in the options (```--subscription='xxxxxxxxx' --tenant='xxxxxxx'
--client-id='xxxxxxxx' --client-secret='xxxxxxxxxx'```).

> The calculated metrics are the total (```--aggregation='total'```) of values on a 900 secondes / 15 min period (```--timeframe='900'```)

Reviewer: ```suggestion
The calculated metrics are the count (```--aggregation='Count'```) of values on a 900 secondes / 15 min period (```--timeframe='900'```)
```

with one sample per 5 minutes (```--interval='PT5M'```).

> This command would trigger a WARNING alarm if the *total jobs* count is reported as over 80

Reviewer: ```suggestion
This command would trigger a WARNING alarm if the number of *requests* is reported as over 800
```

> (```--warning-http-requests='800'```) and a CRITICAL alarm over 90 (```--critical-http-requests='900'```).

Reviewer: ```suggestion
(```--warning-http-requests='800'```) and a CRITICAL alarm over 900 (```--critical-http-requests='900'```).
```
